contracts/exchange/test/transactions_unit_tests.ts

```diff
@@ -357,99 +361,81 @@ blockchainTests.resets('Transaction Unit Tests', ({ provider, web3Wrapper, txDef
             });
             return expect(tx).to.revertWith(expectedError);
         });
-        // FIXME - This should be unskipped when the contracts have been updated to fix this problem.
-        it.skip('should revert if reentrancy occurs in the middle of an executeTransaction call and msg.sender == signer for both calls', async () => {
+        it('should revert if reentrancy occurs in the middle of an executeTransaction call and msg.sender != signer for both calls', async () => {
             const validSignature = randomSignature();
-            const transaction1 = await generateZeroExTransactionAsync({ signerAddress: accounts[0] });
-            const transactionHash1 = transactionHashUtils.getTransactionHashHex(transaction1);
-            const transaction2 = await generateZeroExTransactionAsync({
+            const innerTransaction = await generateZeroExTransactionAsync({ signerAddress: accounts[0] });
+            const innerTransactionHash = transactionHashUtils.getTransactionHashHex(innerTransaction);
+            const outerTransaction = await generateZeroExTransactionAsync({
                 signerAddress: accounts[0],
-                callData: getExecuteTransactionCallData(transaction1, validSignature),
-                returnData: '0xdeadbeef',
+                callData: getExecuteTransactionCallData(innerTransaction, validSignature),
+                returnData: DEADBEEF_RETURN_DATA,
             });
-            const transactionHash2 = transactionHashUtils.getTransactionHashHex(transaction2);
-            const abiEncoder = AbiEncoder.createMethod('TransactionError', ['uint8', 'bytes32']);
-            const errorData = abiEncoder.encode([
-                ExchangeRevertErrors.TransactionErrorCode.NoReentrancy,
-                transactionHash1,
-            ]);
-            const expectedError = new ExchangeRevertErrors.TransactionExecutionError(transactionHash2, errorData);
-            const tx = transactionsContract.executeTransaction.sendTransactionAsync(transaction2, validSignature, {
-                from: accounts[0],
+            const outerTransactionHash = transactionHashUtils.getTransactionHashHex(outerTransaction);
+            const errorData = new ExchangeRevertErrors.TransactionInvalidContextError(
+                innerTransactionHash,
+                accounts[0],
+            ).encode();
+            const expectedError = new ExchangeRevertErrors.TransactionExecutionError(outerTransactionHash, errorData);
+            const tx = transactionsContract.executeTransaction.sendTransactionAsync(outerTransaction, validSignature, {
```

Comment:
These should all be `awaitTransactionSuccessAsync()` calls, even if they revert.